python/_tskitmodule.c

```diff
@@ -9860,6 +9860,150 @@ TreeSequence_divergence_matrix(TreeSequence *self, PyObject *args, PyObject *kwd
     return ret;
 }
 
+/* TODO:
+ * in the default case (no time windows) we calculate statistics at the level of nodes.
```

Comment:
Why don't we do this in Python instead? It should be easy to create a set of time-windows that correspond to the node times using numpy, right? In general, we should only do stuff in C if we really have to, particularly if it involves twisty logic around default input parameters.

Reply:
The time windows thing is tricky. We want to support 3 cases:
1. Per-node output, in which case nodes with the same timepoint are separate rows
2. Per-timepoint output, in which case nodes with the same timepoint are pooled into the same row (and row order is sorted by time)
3. Per-window output, in which case nodes are pooled to windows.

(2) and (3) can be achieved by passing in the right numpy array. However, (1) needs special handling, which is what the `num_time_windows == 0` branch handles.

Reply:
Hmm, ok. Maybe we should have an explicit mode option then or something?

Reply:
That seems like a good idea-- I'll deal with the other issues first, then we can work out the right way to handle "nodes" mode.

Reply:
Yeah, let's get a version with required time windows merged, and figure out the other two cases in a follow up